In [1]:
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC
from sklearn.tree import DecisionTreeClassifier 
from sklearn.ensemble import RandomForestClassifier

import numpy as np
from sklearn.datasets import load_iris
import matplotlib.pyplot as plt

iris = load_iris()

In [2]:
dir(iris)

['DESCR',
 'data',
 'data_module',
 'feature_names',
 'filename',
 'frame',
 'target',
 'target_names']

### K Fold cross validation score

In [3]:
from sklearn.model_selection import cross_val_score
# function for cross validation score
def cross_func(model, cv):
    return cross_val_score(model, iris.data, iris.target, cv=cv)

### Logistic Regression

In [4]:
lr_score = cross_func(LogisticRegression(solver='liblinear', multi_class='ovr'),10)
np.average(lr_score)

0.9533333333333334

### Support Vector Machine

In [5]:
svm_score = cross_func(SVC(gamma='auto'),10)
np.average(svm_score)

0.9800000000000001

### Decision Tree

In [6]:
dt_score = cross_func(DecisionTreeClassifier(criterion='entropy'),10)
np.average(dt_score)

0.96

### Random Forest

In [10]:
rf_score = cross_func(RandomForestClassifier(n_estimators=30),10)
np.average(rf_score)

0.96

In [13]:
rf_score = cross_func(RandomForestClassifier(n_estimators=40),10)
np.average(rf_score)

0.9466666666666667

In [14]:
rf_score = cross_func(RandomForestClassifier(n_estimators=60),10)
np.average(rf_score)

0.96

#### Best score so far is from SVM: 0.98 (or) 98%